zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apiml/ApimlIntegrationFailureDetector.java

```diff
@@ -9,38 +9,33 @@
  */
 package org.zowe.commons.apiml;
 
-import javax.net.ssl.SSLHandshakeException;
-import javax.net.ssl.SSLPeerUnverifiedException;
-
-import org.apache.commons.lang3.exception.ExceptionUtils;
-import org.slf4j.Marker;
-import org.zowe.commons.error.CommonsErrorService;
-import org.zowe.commons.spring.SpringContext;
-
 import ch.qos.logback.classic.Level;
 import ch.qos.logback.classic.Logger;
 import ch.qos.logback.classic.turbo.TurboFilter;
 import ch.qos.logback.core.spi.FilterReply;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Marker;
+import org.zowe.commons.error.CommonsErrorService;
+
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLPeerUnverifiedException;
 
 @Slf4j
 public class ApimlIntegrationFailureDetector extends TurboFilter {
 
     @Override
     public FilterReply decide(Marker marker, Logger logger, Level level, String format, Object[] params, Throwable t) {
-        if (reportFatalErrorAndDecideToExit(level, t)) {
-            System.exit(1);
+        if (reportFatalError(level, t)) {
+            return FilterReply.DENY;
         }
 
         if ((logger != null) && logger.getName().contains("com.netflix")
                 && (logger.getName().contains("DiscoveryClient")
-                        || logger.getName().contains("RedirectingEurekaHttpClient"))) {
+                || logger.getName().contains("RedirectingEurekaHttpClient"))) {
             if (logger.getLevel() == Level.ERROR) {
                 String message = ExceptionUtils.getMessage(t);
-                if (message == null) {
-                    message = ExceptionUtils.getRootCauseMessage(t);
-                }
-                if ((message != null) && !message.isEmpty()) {
+                if (!message.isEmpty()) {
```

Comment:
Are you sure, that getRootCauseMessage cannot return null? I guess it is better to use `!StringUtils.isEmpty(message)`